arch/cortex-m3/src/systick.rs

```diff
@@ -1,8 +1,8 @@
 //! ARM Cortex-M3 SysTick peripheral.
 
 use kernel;
-use kernel::common::regs::{ReadOnly, ReadWrite};
 use kernel::common::StaticRef;
+use tock_regs::regs::{ReadOnly, ReadWrite};
```

Comment:
I think it would be better to have the kernel crate depend on the new tock_regs crate, and re-export the regs they way they are now so we don't need all of these changes.

Reply:
What's the rationale for re-exporting them through the kernel? I'm happy to make the change, but I'm not sure why it would be preferable compared to directly referencing the tock_regs crate. Do we expect all libraries to be re-exported through the kernel in Tock?

Reply:
It would mean not having to change all these files every time we do this (since there is a motion to move the cells too), and it seems like all of these helper functions should come from the same place (and I think it makes sense to stick with them coming from the kernel trait).

Reply:
@bradjc 

Do you mean that only the kernel imports the crate(tock_regs) and then exports as pub?

For example in `lib.rs`:

```rust
extern crate tock_regs;

pub use tock_regs;
```

Then others just use `use kernel::tock_regs::ReadOnly`?!

Reply:
No, the kernel lib.rs does whatever it has to so that chips use `kernel::common::regs::ReadOnly` as they do now.

Reply:
I agree with Brad that centralizing the dependencies within Tock (by re-exporting through the kernel crate) is a good idea.

Reply:
Ah, makes sense!